spec/services/users/manage_credits_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe Users::ManageCredits, type: :service do
+  describe "When the the service is called" do
+    subject(:invoke_service) { described_class.call(user, user_params) }
+
+    let(:user) { create(:user) }
+    let(:organization) { create(:organization) }
+
+    context "when user add the proper amount of credits" do
+      let(:user_params) do
+        { add_credits: "4" }
+      end
+
+      before { invoke_service }
+
+      it { expect(user.reload.credits_count).to eq 4 }
```

Comment:
A better way to test this is to check if the `credits_count` was updated correctly, something like this:

```ruby
it "adds the correct amount of credits to the user" do
 expect {
   invoke_service
 }.to change { user.reload }.by(user_params[:add_credits])
end
```

or

```ruby
it "adds the correct amount of credits to the user" do
 expect {
   invoke_service
 }.to change { user.reload }.to(user_params[:add_credits])
end
``

If it's really necessary to check `unspent_credits_count`, then do it in the same way